src/main/java/com/team4/testingsystem/services/impl/TestEvaluationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.team4.testingsystem.services.impl;
+
+import com.team4.testingsystem.entities.Answer;
+import com.team4.testingsystem.entities.Question;
+import com.team4.testingsystem.entities.Test;
+import com.team4.testingsystem.entities.TestQuestionID;
+import com.team4.testingsystem.services.ChosenOptionService;
+import com.team4.testingsystem.services.TestEvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class TestEvaluationServiceImpl implements TestEvaluationService {
+    private final ChosenOptionService chosenOptionService;
+
+    @Autowired
+    public TestEvaluationServiceImpl(ChosenOptionService chosenOptionService) {
+        this.chosenOptionService = chosenOptionService;
+    }
+
+    public int getEvaluationByTest(Test test) {
```

Comment:
Too complex. It will be something like that: 
return choosenOptionService.getChoosenOptionByTest(test).map(ChoosenOption::getAnswer).filter(Answer::isCorrect) .count();